lib/streamlit/hello.py

```diff
@@ -17,55 +17,172 @@
+    st.markdown("... or maybe with _*markdown*_.")
+
+
+def demo_2():
+    """
+    In this demo, we ask you to enter your name in the input box below.
+    Streamlit will print it out with a number of repetitions given by a
+    slider that you can control.
+    """
+    name = st.text_input("Your name")
+    repetitions = st.slider("Repetitions", 1, 100, 10)
+    st.write(name + "".join((" %s" % name) * (repetitions - 1)))
 
+
+def demo_3():
+    """
+    This demo shows how to use Streamlit to implement a progress bar.
+    """
+    import time
+
+    progress_text = st.text("0%")
     progress_bar = st.progress(0)
-    status_text = st.empty()
-    chart = st.line_chart(np.random.randn(10, 1))
+    success = st.empty()
+    for percent_complete in range(1, 101):
+        progress_text.text("%d%%" % percent_complete)
+        progress_bar.progress(percent_complete)
+        time.sleep(0.1)
+    progress_bar.empty()
+    success.success("Complete!")
+    st.button("Re-run")
 
-    for i in range(1, 101):
-        # Update progress bar.
-        progress_bar.progress(i)
 
-        new_rows = np.random.randn(10, 1)
+def demo_4():
+    """
+    Use the slider in the sidebar to mix your favorite color up.
+    """
+    red = st.sidebar.slider("Red", 0, 255, 0)
+    green = st.sidebar.slider("Green", 0, 255, 128)
+    blue = st.sidebar.slider("Blue", 0, 255, 0)
+    array = np.zeros([512, 512, 3], dtype=np.uint8)
+    array[:, :, 0].fill(red)
+    array[:, :, 1].fill(green)
+    array[:, :, 2].fill(blue)
+    st.markdown("#### RGB color (%d, %d, %d)" % (red, green, blue))
+    st.image(array)
 
-        # Update status text.
-        status_text.text("The latest random number is: %s" % new_rows[-1, 0])
 
-        # Append data to the chart.
-        chart.add_rows(new_rows)
+# Data for demo_5 derived from
+# https://www.kaggle.com/rounakbanik/the-movies-dataset
+DATASET_URL = (
+    "https://streamlit-demo-data.s3-us-west-2.amazonaws.com/movies-revenue.csv.gz"
+)
 
-        # Pretend we're doing some computation that takes time.
-        time.sleep(0.1)
 
-    status_text.text("Done!")
-    st.balloons()
+def demo_5():
+    """
+    Discover the gross revenue of a movie of your liking!
+    Note that the network loading and preprocessing of the data is cached
+    by using Streamlit st.cache annotation.
+    """
+    import pandas as pd
+
+    @st.cache
+    def load_dataframe_from_url():
+        return pd.read_csv(DATASET_URL).transform(
+            {"title": lambda x: x, "revenue": lambda x: round(x / 1000 / 1000)}
+        )
+
+    df = load_dataframe_from_url()
+    movie = st.text_input("Search movie titles")
+    df = df[df.title.str.contains(movie, case=False)].sort_values(
+        ascending=False, by="revenue"
+    )
+    st.markdown("#### Result dataframe for %d movie(s)" % df.shape[0])
+    st.dataframe(df.rename(columns={"title": "Title", "revenue": "Revenue [$1M]"}))
+
 
+DEMOS = OrderedDict(
+    {
+        "---": intro,
```

Comment:
Maybe  we can remove `Intro` and just show them the `Basic Text` demo on first page load?